yt_dlp/extractor/cnn.py

```diff
@@ -1,146 +1,225 @@
+                query={
+                    'id': media_id,
+                    'stellarUri': parent_uri,
+                })
+            for direct_url in traverse_obj(video_data, ('files', ..., 'fileUri', {url_or_none})):
+                resolution, bitrate = None, None
+                if mobj := re.search(r'-(?P<res>\d+x\d+)_(?P<tbr>\d+)k\.mp4', direct_url):
+                    resolution, bitrate = mobj.group('res', 'tbr')
+                formats.append({
+                    'url': direct_url,
+                    'format_id': 'direct',
+                    'quality': 1,
+                    'tbr': int_or_none(bitrate),
+                    **parse_resolution(resolution),
+                })
+            for sub_data in traverse_obj(video_data, (
+                    'closedCaptions', 'types', lambda _, v: url_or_none(v['track']['url']), 'track')):
+                subtitles.setdefault(sub_data.get('lang') or 'en', []).append({
+                    'url': sub_data['url'],
+                    'name': sub_data.get('label'),
+                })
 
-    def _real_extract(self, url):
-        webpage = self._download_webpage(url, url_basename(url))
-        cnn_url = self._html_search_regex(r'data-url="(.+?)"', webpage, 'cnn url')
-        return self.url_result(cnn_url, CNNIE.ie_key())
+            if app_id:
+                media_data = self._download_json(
+                    f'https://medium.ngtv.io/v2/media/{media_id}/desktop', media_id, fatal=False,
+                    query={'appId': app_id})
+                m3u8_url = traverse_obj(media_data, (
+                    'media', 'desktop', 'unprotected', 'unencrypted', 'url', {url_or_none}))
+                if m3u8_url:
+                    fmts, subs = self._extract_m3u8_formats_and_subtitles(
+                        m3u8_url, media_id, 'mp4', m3u8_id='hls', fatal=False)
+                    formats.extend(fmts)
+                    self._merge_subtitles(subs, target=subtitles)
 
+            entries.append({
+                **traverse_obj(player_data, {
+                    'title': ('data-headline', {clean_html}),
+                    'description': ('data-description', {clean_html}),
+                    'duration': ('data-duration', {parse_duration}),
+                    'timestamp': ('data-publish-date', {parse_iso8601}),
+                    'thumbnail': (
+                        'data-poster-image-override', {json.loads}, 'big', 'uri', {url_or_none},
+                        {functools.partial(update_url, query='c=original')}),
+                    'display_id': 'data-video-slug',
+                }),
+                **traverse_obj(video_data, {
+                    'timestamp': ('dateCreated', 'uts', {float_or_none(scale=1000)}),
+                    'description': ('description', {clean_html}),
+                    'title': ('headline', {str}),
+                    'modified_timestamp': ('lastModified', 'uts', {float_or_none(scale=1000)}),
```

Comment:
I'd use `int_or_none` for the timestamps here to stay consistent with the fallback values
```suggestion
                    'timestamp': ('dateCreated', 'uts', {int_or_none(scale=1000)}),
                    'description': ('description', {clean_html}),
                    'title': ('headline', {str}),
                    'modified_timestamp': ('lastModified', 'uts', {int_or_none(scale=1000)}),
```